pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -429,7 +429,11 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 		if err != nil {
 			return ctrl.syncStatusOnly(cfg, err, "could not generate the original Kubelet config: %v", err)
 		}
-		dataURL, err := dataurl.DecodeString(originalKubeletIgn.Contents.Source)
+		sourceData := ctrlcommon.StrFromStrPtr(originalKubeletIgn.Contents.Source)
+		if len(sourceData) == 0 {
+			sourceData = "data:,"
+		}
+		dataURL, err := dataurl.DecodeString(sourceData)
```

Comment:
use `*originalKubeletIgn.Contents.Source` directly as you did in https://github.com/openshift/machine-config-operator/pull/583/commits/15ea796b9345b9ef15834c6e6309e97da82af420#diff-22412a83973655ff84984dc7d9969013R603

Reply:
based on what i accidentally ran thru this yesterday, if you give dataurl.DecodeString an empty string, it will throw an error. is that desirable behavior in this case?

Reply:
yeah, I guess we need to add the checks for this to the place I linked as well then right?

Reply:
ahhhh yes good catch!

Reply:
done